libraries/orchestrator.rb

```diff
@@ -0,0 +1,276 @@
+require 'poise'
+require 'chef/resource'
+require 'chef/provider'
+
+class Chef
+  class Resource
+    # Orchestrator service installation and configuration resource
+    class OrchestratorService < Chef::Resource
+      include Poise
+      require 'securerandom'
+
+      provides(:orchestrator_service)
+      actions(:install)
+
+      attribute(
+        :database_backend,
+        kind_of: String,
+        default: lazy { 'mysql' }
+      )
+      attribute(
+        :install_mysql,
+        kind_of: [TrueClass, FalseClass],
+        default: true
+      )
+      attribute(
+        :mysql_root_password,
+        kind_of: String,
+        default: lazy { random_password }
+      )
+      attribute(
+        :orchestrator_database_name,
+        kind_of: String,
+        default: lazy { node['orchestrator']['database']['name'] }
+      )
+      attribute(
+        :orchestrator_database_user,
+        kind_of: String,
+        default: lazy { node['orchestrator']['database']['user'] }
+      )
+      attribute(
+        :orchestrator_database_password,
+        kind_of: String,
+        default: lazy { 'secret' }
+      )
+      attribute(
+        :mysql_topology_user,
+        kind_of: String,
+        default: lazy { 'topology' }
+      )
+      attribute(
+        :mysql_topology_password,
+        kind_of: String,
+        default: lazy { 'secret' }
+      )
+      attribute(
+        :http_user,
+        kind_of: String,
+        default: lazy { 'http_user' }
+      )
+      attribute(
+        :http_password,
+        kind_of: String,
+        default: lazy { 'secret' }
+      )
+      attribute(
+        :raft_nodes,
+        kind_of: Array,
+        default: []
+      )
+
+      def random_password
+        SecureRandom.hex
+      end
+    end
+  end
+
+  class Provider
+    # Orchestrator service installation and configuration resource
+    # rubocop:disable Metrics/ClassLength
+    class OrchestratorService < Chef::Provider
+      include Poise
+
+      provides(:orchestrator_service)
+
+      # rubocop:disable Metrics/AbcSize
+      def action_install
+        orchestrator_install
+        orchestrator_system_user
+        create_raft_directory
+        mysql_db_install if install_mysql?
+        mysql_db_setup if new_resource.database_backend == 'mysql'
+        sqlite_db_install if new_resource.database_backend == 'sqlite'
+        orchestrator_config
+        orchestrator_service
+      end
+      # rubocop:enable Metrics/AbcSize
+
+      def install_mysql?
+        return true if new_resource.install_mysql && new_resource.database_backend == 'mysql'
+
+        false
```

Comment:
you can remove false, function will return nil == false, by default

```
$ irb
irb(main):001:0> def true_or?
irb(main):002:1> return true if false
irb(main):003:1> end
=> :true_or?
irb(main):004:0> true_or?
=> nil
```

Reply:
@ernestas-vinted  - fixed